### PR TITLE
Moving Chloropleth data manipulation to data provider

### PR DESCRIPTION
**Pull request by kkatzen**

### CODE REVIEW

File: app/src/features/reports/TellMeAboutReport.tsx [hunk 1]
```diff
@@ -93,48 +96,59 @@ function TellMeAboutReport(props: { variable: VariableId }) {
...
+                signalListeners={signalListeners}
+                varField={props.variable}
+                legendTitle={variableProvider.variableName}
+                data={dataset}
+              />
+              <p>
+                Click on some states to see data in this table, shift click on
+                map to reset.
+              </p>
+              {CountyLevelTable(countyList, variableProvider.variableName)}
```
Comment (aaronsn) [inline app/src/features/reports/TellMeAboutReport.tsx:141]: Can this be replaced with the TableChart component?

File: app/src/features/reports/TellMeAboutReport.tsx [hunk 2]
```diff
@@ -93,48 +96,59 @@ function TellMeAboutReport(props: { variable: VariableId }) {
...
+            </Grid>
+            <Grid item xs={12} sm={12} md={6} className={styles.PaddedGrid}>
+              <TableChart
+                data={dataset}
+                columns={["state_name", props.variable]}
```
Comment (aaronsn) [inline app/src/features/reports/TellMeAboutReport.tsx:146]: Can you use the variableProvider.variableName for the column display name?

File: app/src/features/reports/TellMeAboutReport.tsx [hunk 3]
```diff
@@ -93,48 +96,59 @@ function TellMeAboutReport(props: { variable: VariableId }) {
...
-            </p>
-            {CountyLevelTable(countyList, variableProvider.variableName)}
+      {() => {
+        const breakdowns =
+          race === THE_USA_STRING
+            ? Breakdowns.byState()
+            : Breakdowns.byState().andRace();
+        let dataset = variableProvider
+          .getData(datasetStore.datasets, breakdowns)
+          .filter((r) => race === THE_USA_STRING || r.race === race);
```
Comment (aaronsn) [inline app/src/features/reports/TellMeAboutReport.tsx:106]: Why is this filtering using an OR? Shouldn't it just be r.race === race and rely on the race state?

**Review by aaronsn — APPROVED** [review]
This is great, thanks!

File: app/src/features/charts/UsaChloroplethMap.tsx [hunk 4]
```diff
@@ -96,25 +67,16 @@ function UsaChloroplethMap(props: {
...
     setSpec({
       $schema: "https://vega.github.io/schema/vega/v5.json",
       description:
         "A choropleth map depicting U.S. diabetesloyment temp_maxs by county in 2009.",
       data: [
-        varDataset,
-
+        {
+          name: VAR_DATASET,
+          transform: varTransformer,
```
[the remark this answers is not in the document]
Comment (kkatzen) [reply app/src/features/charts/UsaChloroplethMap.tsx:77]: oops! good catch

File: app/src/features/reports/TellMeAboutReport.tsx (hunk 1 above)
Comment (kkatzen) [reply app/src/features/reports/TellMeAboutReport.tsx:141]: It's not totally straightforward- I'll follow up in another PR

**Pull request merged**